box2dbody.cpp

```diff
@@ -263,6 +263,9 @@ void Box2DBody::createBody()
         return;
     }
 
+    if(!mTarget)
+        mTarget = static_cast<QQuickItem *>(parent());
```

Comment:
You can't just `static_cast` any parent because there is no guarantee that it is a `QQuickItem`. You can put the `Body` instance anywhere after all. Probably it would be better to use `qobject_cast`.

Also please put a space between `if` and `(`. :-)